teuthology/scrape.py

```diff
@@ -0,0 +1,516 @@
+# Origin: https://github.com/jcsp/scrape/blob/master/scrape.py
+# Author: John Spray (github.com/jcsp)
+
+import difflib
+from errno import ENOENT
+from gzip import GzipFile
+import sys
+import os
+import yaml
+from collections import defaultdict
+import re
+import logging
+import subprocess
+
+import six
+
+log = logging.getLogger('scrape')
+log.addHandler(logging.StreamHandler())
+log.setLevel(logging.INFO)
+
+
+class Reason(object):
+    def get_description(self):
+        return self.description
+
+    def get_detail(self):
+        return None
+
+
+def grep(path, expr):
+    """
+    Call out to native grep rather than feeding massive log files through python line by line
+    """
+    p = subprocess.Popen(["grep", expr, path], stdout=subprocess.PIPE)
+    p.wait()
+    out, err = p.communicate()
+    out = six.ensure_str(out)
```

Comment:
Sorry for asking this, but self modifying variables is not a good practice, can you just put this:
```
if p.returncode:
    return []
else:
    return six.ensure_str(out).split("\n")
```

Reply:
I'll keep in mind from the next time. Fixing this right away.